### PR TITLE
ssh/tailssh: add support for forwarding unix sockets

### LINKED ISSUE #6232: Remote port does not work for unix socket (Tailscale SSH)
### What is the issue?

When using the [`RemoteForward`](https://manpages.debian.org/bullseye/openssh-client/ssh_config.5.en.html#RemoteForward) with a Unix domain socket over Tailscale SSH it fails with the following error given the following config:

Config:

```
RemoteForward /tmp/test.socket /tmp/test.socket
```

Error on connect:
```
Warning: remote port forwarding failed for listen path /tmp/test.socket
```

### Steps to reproduce

_No response_

### Are there any recent changes that introduced the issue?

_No response_

### OS

Linux, macOS

### OS version

_No response_

### Tailscale version

1.32.2

### Bug report

_No response_

### PR DESCRIPTION
**Pull request by Xenfo**
closes #6232 

Based off of @deansheather's work at https://github.com/gliderlabs/ssh/pull/196 and https://github.com/coder/coder/blob/2c0f653aa85ab4d2c4be410642776c772eb524ac/agent/agentssh/forward.go, this PR finally adds support for forwarding unix sockets. This allows workflows such as GPG agent forwarding to work.

Just as a side note, I'm not very familiar with networking and the security that comes with it so this code will probably need to be modified to meet whatever requirements Tailscale has during review.

### CODE REVIEW

File: tailcfg/tailcfg.go [hunk 1]
```diff
@@ -2456,6 +2457,14 @@ type SSHAction struct {
...
 	AllowRemotePortForwarding bool `json:"allowRemotePortForwarding,omitempty"`
 
+	// AllowLocalUnixForwarding, if true, allows accepted connections
+	// to use local unix forwarding if requested.
+	AllowLocalUnixForwarding bool `json:"allowLocalUnixForwarding,omitempty"`
+
+	// AllowRemoteUnixForwarding, if true, allows accepted connections
+	// to use remote unix forwarding if requested.
+	AllowRemoteUnixForwarding bool `json:"allowRemoteUnixForwarding,omitempty"`
+
```
Comment (Xenfo) [inline tailcfg/tailcfg.go:2467]: Not sure how to actually change these values on the client, could it be part of another project? Currently they seem to default to false.

File: tempfork/gliderlabs/ssh/streamlocal.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+			return false, nil
+		}
+
+		ln, err := net.Listen("unix", addr)
+		if err != nil {
+			// TODO: log unix listen failure
+			return false, nil
+		}
+
+		if err := os.Chmod(addr, os.FileMode(0777)); err != nil {
```
Comment (Xenfo) [inline tempfork/gliderlabs/ssh/streamlocal.go:146]: Is there a better permission that can be used on the socket?

**Comment by oxtoacart** [comment]
@Xenfo Thank you for submitting this. We don't want our fork of `gliderlabs/ssh` to diverge too far from upstream, and there's actually the possibility that we'll unfork it completely. Given this, we'll need this feature merged upstream to `gliderlabs/ssh` before we can take it. Once it is, I'd love to help with rebasing our fork to the latest upstream.

**Comment by t4vvd** [comment]
Please correct me if I think open SSH is needed only for private running repositories and not public repositories. 
For my Deriv trading the Volatility 10 Index on the Deriv API and Explorer can anyone explain ? 
Thanks PaulM.

**Pull request closed without merging**